app/models/product_import/entry_validator.rb

```diff
@@ -323,8 +323,10 @@ def validate_inventory_item(entry, variant_override)
 
     def create_inventory_item(entry, existing_variant)
       variant_override = find_or_initialize_variant_override(entry, existing_variant)
-      variant_override.assign_attributes(count_on_hand: entry.on_hand, import_date: @import_time)
-      check_on_hand_nil(entry, variant_override)
+
+      check_variant_override_stock_settings(entry, variant_override)
```

Comment:
I find this very confusing.

1. count_on_hand and on_demand of the existing variant override are assigned based on entry.
2. entry.on_hand_nil is assigned based on the the variant override.
3. on_demand of the variant override is assigned again based on entry.

In this context, the entry is only used to provide `price` and `on_demand` for the variant_override. So what if `check_variant_override_stock_settings` was called `variant_override_attributes` and returned the right hash based on entry and existing variant override?

I guess that doesn't sanitise `entry` which could be used somewhere else. Am I right with my assumptions? Is there a way to avoid these circular assignments? I would find it easier if it was more linear, for example:

1. Derive truth from input variables, for example create hash from entry and override.
2. Assign values to variant override (which is concerned here).

Or if the entry needs to be modified:

1. Sanitise entry with help of variant override.
2. Assign values to variant override.

Sorry if this is confusing. I'm not sure if it's understandable what I'm trying to say. :-D.

Modifying state within objects is always more dangerous than having immutable data structures.

Reply:
I actually based this on the `check_on_hand_nil` method which it was originally using. I read through some of the import related code, but I confess I don't know how exactly `entry.on_hand_nil` is used. However, the attribute was already being set even before this PR: https://github.com/openfoodfoundation/openfoodnetwork/blob/df51d137dc87a819b3646e501798079ac387046e/app/models/product_import/entry_validator.rb#L354

`check_on_hand_nil` was setting presumed attributes for the VO if needed, and setting `entry.on_hand_nil` if there is no `on_hand` value.

@Matt-Yorkley Does this change seem safe enough to you?

If yes, @mkllnk are you okay with keeping that part as is for now?

Reply:
Yes, I'm okay with it.